public/scripts/services/date.js

```diff
@@ -2,603 +2,20 @@
-            days = {
-                'sun': 0,
-                'mon': 1,
-                'tue': 2,
-                'wed': 3,
-                'thu': 4,
-                'fri': 5,
-                'sat': 6
-            }
-            ranges = {
-                'yea': 'FullYear',
-                'mon': 'Month',
-                'day': 'Date',
-                'hou': 'Hours',
-                'min': 'Minutes',
-                'sec': 'Seconds'
-            }
-            function lastNext (type, range, modifier) {
-                var diff
-                var day = days[range]
-                if (typeof day !== 'undefined') {
-                    diff = day - date.getDay()
-                    if (diff === 0) {
-                        diff = 7 * modifier
-                    } else if (diff > 0 && type === 'last') {
-                        diff -= 7
-                    } else if (diff < 0 && type === 'next') {
-                        diff += 7
-                    }
-                    date.setDate(date.getDate() + diff)
-                }
-            }
-            function process (val) {
-                // @todo: Reconcile this with regex using \s, taking into account
-                // browser issues with split and regexes
-                var splt = val.split(' ')
-                var type = splt[0]
-                var range = splt[1].substring(0, 3)
-                var typeIsNumber = /\d+/.test(type)
-                var ago = splt[2] === 'ago'
-                var num = (type === 'last' ? -1 : 1) * (ago ? -1 : 1)
-                if (typeIsNumber) {
-                    num *= parseInt(type, 10)
-                }
-                if (ranges.hasOwnProperty(range) && !splt[1].match(/^mon(day|\.)?$/i)) {
-                    return date['set' + ranges[range]](date['get' + ranges[range]]() + num)
-                }
-                if (range === 'wee') {
-                    return date.setDate(date.getDate() + (num * 7))
-                }
-                if (type === 'next' || type === 'last') {
-                    lastNext(type, range, num)
-                } else if (!typeIsNumber) {
-                    return false
-                }
-                return true
-            }
-            times = '(years?|months?|weeks?|days?|hours?|minutes?|min|seconds?|sec' +
-                '|sunday|sun\\.?|monday|mon\\.?|tuesday|tue\\.?|wednesday|wed\\.?' +
-                '|thursday|thu\\.?|friday|fri\\.?|saturday|sat\\.?)'
-            regex = '([+-]?\\d+\\s' + times + '|' + '(last|next)\\s' + times + ')(\\sago)?'
-            match = text.match(new RegExp(regex, 'gi'))
-            if (!match) {
-                return fail
-            }
-            for (i = 0, len = match.length; i < len; i++) {
-                if (!process(match[i])) {
-                    return fail
-                }
-            }
-            return (date.getTime() / 1000)
+            return new Intl.DateTimeFormat('en-US', {
```

Comment:
Does this force the locale to `en-US`? If so, should we be doing that?

Reply:
@abnegate @TorstenDittmann for you please